### DISCUSSION TITLE
How do I add thumbnail images to the blog posts?

### DISCUSSION THREAD

**Discussion by reignbeox** [timlrx/tailwind-nextjs-starter-blog#256] [Q&A] [answered]
I'm really sorry that I have to make this post but I'm a noob in website stuff and am trying to fumble through this.

For two and a half weekends now I've been trying to get thumbnail images to show up on the blog page like in [this example from the recipe blog](https://tambouille.vercel.app/).

I understand that the index.js says how stuff is rendered on the page and looking at the tambouille-blog code I tried my hardest to copy bit by bit but I always get some sort of error. Occasionally I manage to fix them but then ultimately get stuck at some point and need to revert.
Nonetheless, I can't imagine it to be that much of a hassle. It's just about adding a thumbnail after all no?

Don't really know who to ask or where to look for answers except here.

tldr: how do I add a thumbnail image to the blog post list?
like so:
![Capture2](https://user-images.githubusercontent.com/23138223/136676734-4acfa7ce-c670-45e6-a0e0-7dcd369d3e05.png)

**Comment by vahidtakro** [comment, marked as answer]
> tldr: how do I add a thumbnail image to the blog post list? 

simply add these lines to your `ListLayout.js` / `index.js` file after `</time></dd>` line:

```
    {images ? (
      <Image
        className="rounded object-cover"
        src={`/_next/image?url=${images}&w=640&q=20`}
        slug="slug"
        title={`${title}`}
        alt={`${title}`}
        layout="responsive"
        placeholder="blur"
        blurDataURL={`/_next/image?url=${images}&w=16&q=1`}
        width={640}
        height={400}
      />
    ) : null}
```

you should also include `images` into your frontMatter:
`const { slug, date, title, summary, tags, images } = frontMatter`

and import nextjs Image Component at top of the page(s):
`import Image from 'next/image'`

for more information: [next/image](https://nextjs.org/docs/api-reference/next/image)

**Reply by reignbeox** [reply]
Figured it was something as simple as this
Tho I hoped to understand the code of the other blog more, but I don't :D

But I now understand that index.js and listlayout.js can be independent of one another!

Thank you sir o7

Would you also perhaps know where I can edit the padding inside of the containers? The image sits a bit too snug on the text.

**Reply by aloisdg** [reply]
@reignbeox The code on the recipe blog is from a fork of this starter pre-v1. It can a bit clunky.

```jsx
  let thumbnail = (
    <Link href={`/recettes/${slug}`} title={title} className="w-full xl:w-auto">
      <Image
        alt="illustration"
        className="rounded object-cover"
        src={image}
        layout="responsive"
        width={640}
        height={400}
      />
    </Link>
  ```

The exact code can be found [here](https://github.com/aloisdg/tambouille/blob/bc7b84c27327ba55cfbe43e618e49f6f42da3950/components/Article.js#L21)

**Reply by reignbeox** [reply]
@aloisdg Ah I see. makes a bit more sense to me now then.
Actually, have a copy of your project right here trying to decipher parts and bits of it.

Do you remember where you set the padding for the images?
Every value that I change either changes the spacing between the articles or the length of the right-hand text 

![Capture2](https://user-images.githubusercontent.com/23138223/136698226-9994416a-0324-4c60-97d3-2f6928389cc3.png)
/

**Reply by reignbeox** [reply]
Ah I got it. gap-2!

**Reply by aloisdg** [reply]
For rwd purpose, I use `gap-2 xl:gap-4` :) [source](https://github.com/aloisdg/tambouille/blob/bc7b84c27327ba55cfbe43e618e49f6f42da3950/components/Article.js#L49)

**Reply by vahidtakro** [reply]
> Would you also perhaps know where I can edit the padding inside of the containers? The image sits a bit too snug on the text.

You may need to read how [Tailwind](https://tailwindcss.com) works. so then you can modify the "className" in the way you want.

**Reply by reignbeox** [reply]
> > Would you also perhaps know where I can edit the padding inside of the containers? The image sits a bit too snug on the text.
> 
> You may need to read how [Tailwind](https://tailwindcss.com) works. so then you can modify the "className" in the way you want.

I agree. Now that I managed to fix this one issue, I started looking into other resources to learn from.
One of course is the tailwind documentation but I also found the [tailwind examples](https://tailwindcomponents.com/) where stuff is a little bit more descriptive. 

Thanks for the help people!
